base/sparse/sparsematrix.jl

```diff
@@ -120,7 +120,103 @@ column. In conjunction with [`nonzeros`](@ref) and
 """
 nzrange(S::SparseMatrixCSC, col::Integer) = S.colptr[col]:(S.colptr[col+1]-1)
 
+
+module SparseArrayValidity
+    @enum(_SparseArrayValidity,
+        VALID,
+        COLPTR_LENGTH,
+        COLPTR_FIRST_VAL,
+        NZVAL_LENGTH,
+        COLPTR_SORTED,
+        ROWVAL_LENGTH,
+        ROWVAL_RANGE,
+        ROWVAL_SORTED_COLUMN,
+    )
+end
+
+
+"""
+    checkvalid(::Type{Bool}, S::SparseMatrixCSC; full = true) -> Bool
+
+Return `true` if the sparse matrix `S` has a valid internal representation.
+If `full` is `false` only O(1) checks are done, otherwise full
+O(nnz) checks are performed.
+
+# Examples
+
```

Comment:
(Extra empty line under the seemingly just-established convention, but could be fixed in the dedicated PR instead.)